core/src/main/java/com/graphhopper/routing/AbstractBidirAlgo.java

```diff
@@ -88,6 +96,24 @@ protected void initCollections(int size) {
 
     @Override
     public Path calcPath(int from, int to) {
+        return calcPath(from, to, ANY_EDGE, ANY_EDGE);
+    }
+
+    /**
+     * like {@link #calcPath(int, int)}, but this method also allows to strictly restrict the edge the
+     * path will begin with and the edge it will end with.
+     *
+     * @param fromOutEdge the edge id of the first edge of the path. using {@link EdgeIterator.ANY_EDGE} means
+     *                    not enforcing the first edge of the path
+     * @param toInEdge    the edge id of the last edge of the path. using {@link EdgeIterator.ANY_EDGE} means
+     *                    not enforcing the last edge of the path
+     */
+    public Path calcPath(int from, int to, int fromOutEdge, int toInEdge) {
```

Comment:
In principle we could also move this to the `RoutingAlgorithm`, but for now it was easier to implement this only for bidirectional algorithms. Not sure if we really need this for the others as well and we can add this later if needed (?)